harmony_netcdf_to_zarr/mosaic_utilities.py

```diff
@@ -384,13 +494,25 @@ def get_grid_values(input_values: np.ndarray,
         the input array to the maximum. The grid spacing will be the supplied
         resolution.
 
+        The scale factor is used to ensure the `np.linspace` output does not
+        include recurring decimals. For example, the GPM/IMERG longitude grid
+        output dimension previously included -179.8499999999, instead of
+        -179.85.
+
     """
+    _, scale_factor = scale_to_integers(input_values)
+
     grid_max = input_values.max()
     grid_min = input_values.min()
 
     if grid_min != grid_max:
-        n_grid_points = int(((grid_max - grid_min) / grid_resolution) + 1)
+        n_grid_points = int(
+            np.round(((grid_max - grid_min) / grid_resolution))
+        ) + 1
```

Comment:
The change above (to do `int(np.round(<value>))` rather than `int(<value>)` is important: `int(1.9999999) = 1`, but `int(np.round(1.9999999)) = 2`, which is what we want. This stems, again, from Python sometimes giving values that have recurring decimals, instead of being the expected round numbers.

Apologies for not catching that the first time around.

Reply:
Glad we got this rectified. :-)